static/js/tools/choropleth/choropleth.tsx

```diff
@@ -80,12 +81,17 @@ class ChoroplethMap extends Component<PropsType, StateType> {
 
     Promise.all([geoPromise, valuePromise]).then(
       (values) => {
+        // Coordinates for the map.
         const geoJson = values[0].data[0];
-        const geoIdToValues = values[1].data[0];
+        // All the timeseries data for all places.
+        const data = values[1].data[0];
+        // The latest value for all places.
+        const geoIdToValues = this.filterByDate(data, "latest");
```

Comment:
rather than displaying latest, let's just show the latest date in the drop down selected by default

Reply:
"Latest" shows the latest date for each region individually.

Selecting "2020" would only show those regions with data from 2020.
Selecting "latest" would show the region's latest date, some regions would display data from 2019 others from 2020. Whatever the most up-to-date data is.

They have different functionality, would you want to get rid of "latest" functionality? Or maybe rename it?
Imagine one wants to see the map of covid that gets updated every day. This is a way of displaying all at once. Some updated yesterday, others updated today.

Reply:
Latest date to me implies the next option (in the screenshot, 2019). We could also get widely variable ranges of dates - e.g. would it make sense to compare 2010 data with 2019? It is a correctness issue in my mind, and drawing a chart comparing data across dates is misleading.

@jeffreyoldham wdyt?

Separately, have you tested the behavior for some stat vars with very long histories? Covid, BLS are some that come to mind.

Reply:
![image](https://user-images.githubusercontent.com/11444894/94045436-7736e300-fd9d-11ea-9ce2-e3e4c236b8de.png)
![image](https://user-images.githubusercontent.com/11444894/94045446-7900a680-fd9d-11ea-9e08-efdf6ff14ed6.png)

Long history works fine! 👍 🥇

Reply:
@beets @jeffreyoldham

Reply:
I believe:
1) Users almost always want data from every state (or every country or every county).  They do not want to have to patch together data from different dates to see the overall picture.
2) In the real world, it is unlikely to get data from every entity at the same time.  E.g., each country conducts population censuses at different times.  Some are several years old.  We are spoiled with getting time-consistent per-US state data from US federal agencies, but we cannot expect the same for some data, e.g., counts of drivers licenses.

Thus, I think that users will be OK with having "latest" show data of different dates as long as:
1) it really is the latest data for that state and
2) the time gap for all the displayed data is not too large.  E.g., if one is looking at daily data, I think it would be hard to complain that some data is from Monday while others is from Friday.  However, if it is from a month ago, it might be too old to display.  If one is looking at monthly data, if a state's data is more than a year older than some other state's data, perhaps we should not show it.  If annual data, perhaps a decade gap may be too large.

In practice, I do not expect the gap between the newest and oldest data to be too large for US state level data, but the real world may disprove me.

Would it work to add a todo to the code?  We have identified an issue.  The feature seems like an improvement.  We expect we will produce good results almost all of the time, but there is a possibility we may not.  Should we address it in a separate pull request focussed on the time-data disparity issue?

Reply:
Thanks Jeffrey - that sounds good to me. I agree it is probably only a few corner cases where this might not be what users are looking for.

Sure, let's add a TODO to show the range of dates somewhere e.g. "showing data from 2010-2011"